.travis.yml

```diff
@@ -1,17 +1,47 @@
 language: python
 sudo: false
-python:
-    - 2.6
-    - 2.7
-    - 3.2
-    - 3.3
-    - 3.4
-    - 3.5
-    - pypy
-    - pypy3
+matrix:
+    include:
+        - os: linux
+          python: 2.6
+        - os: linux
+          python: 2.7
+        - os: linux
+          python: 3.3
+        - os: linux
+          python: 3.4
+        - os: linux
+          python: 3.5
+        - os: linux
+          python: pypy
+        - os: linux
+          python: pypy3
+        - os: osx
+          language: generic
+          env: TERRYFY_PYTHON='homebrew 2'
+        - os: osx
+          language: generic
+          env: TERRYFY_PYTHON='homebrew 3'
+before_install:
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then git clone https://github.com/MacPython/terryfy; fi
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then source terryfy/travis_tools.sh; fi
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then get_python_environment $TERRYFY_PYTHON venv; fi
+    - if [[ "$TERRYFY_PYTHON" == "homebrew 3" ]]; then alias pip=`which pip3` ; fi
 install:
     - pip install .
 script:
     - python setup.py test -q
 notifications:
     email: false
+after_success:
+    - echo [distutils]                                  > ~/.pypirc
+    - echo index-servers = pypi                        >> ~/.pypirc
+    - echo [pypi]                                      >> ~/.pypirc
+    - echo repository=https://pypi.python.org/pypi     >> ~/.pypirc
+    - echo username=USER                               >> ~/.pypirc
+    - echo password=$PYPIPASSWORD                      >> ~/.pypirc
```

Comment:
Won't this expose the PyPI password in the Travis build log?

I hope Travis is smart enough not to echo decrypted variable values when they're used on the command line, but I'd love to see an existing project that uploads stuff to PyPI and see with my own eyes how it actually works.